lib/api/core/validation/index.js

```diff
@@ -800,9 +782,9 @@ function manageErrorMessage(errorContext, errorHolder, message, structured) {
         errorHolder.fieldScope = {};
       }
 
-      pointer = errorHolder.fieldScope;
+      let pointer = errorHolder.fieldScope;
 
-      for (i = 0; i < errorContext.length; i++) {
+      for (let i = 0; i < errorContext.length; i++) {
         if (!pointer.children) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ANestedIfDepth)